### PR TITLE
[Marksmanship] Trueshot suggestions, talents and tier21.

### PR DESCRIPTION
**Pull request by Pewtro**
Hopefully last PR for Marksmanship that's this large for some time. 
This includes sweeping changes to Trueshot module so it once again includes some suggestions.
Adds Natural Mending, Trailblazer, Serpent Sting statistics. 
Updates tier 21 to current iterations. 
Updates Hunter's Mark module to include some sort of precast prediction - aswell as fixing the energize events properly.

### CODE REVIEW

File: src/Parser/Hunter/Shared/Modules/Talents/NaturalMending.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      }
+    }
+  }
+
+  on_byPlayer_cast(event) {
+    if (!event || !event.classResources || event.classResources[0].cost === 0) {
+      return;
+    }
+    this.lastFocusCost = event.classResources[0].cost || 0;
+    const COOLDOWN_REDUCTION_MS = this.cdrPerFocus * this.lastFocusCost;
```
Comment (Gebuz) [inline src/Parser/Hunter/Shared/Modules/Talents/NaturalMending.js]: local constants should be named like variables, so in this case `cooldownReductionMS`

File: src/Parser/Hunter/Shared/Modules/Talents/NaturalMending.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    if (this.spellUsable.isOnCooldown(SPELLS.EXHILARATION.id)) {
```
Comment (Gebuz) [inline src/Parser/Hunter/Shared/Modules/Talents/NaturalMending.js]: you could avoid the nested if statements by checking for the opposite and returning early (after doing `this.wastedExhilReductionMs += COOLDOWN_REDUCTION_MS;`

File: src/Parser/Hunter/Marksmanship/Modules/Abilities.js [hunk 3]
```diff
@@ -65,6 +65,11 @@ class Abilities extends CoreAbilities {
           ),
         },
       },
+      {
+        spell: SPELLS.EXPLOSIVE_SHOT_DETONATION,
```
[the remark this answers is not in the document]
Comment (Pewtro) [reply src/Parser/Hunter/Marksmanship/Modules/Abilities.js]: It has no GCD and has no CD. It's the detonate of the Explosive Shot, and can be used at any given time after casting Explosive Shot and it's in the air. Just thought it interesting to see if people did activate it, I'll look into AoECastEfficiency.

[the remark this answers is not in the document]
Comment (Gebuz) [reply src/Parser/Hunter/Marksmanship/Modules/Abilities.js]: ah my bad, I didn't realize it was the detonation part of the ability.

**Pull request merged**